### PR TITLE
CIS-115 Test old iOS SDKs nightly

### PR DESCRIPTION
**Pull request by b-onc**
Test run here: https://github.com/GetStream/stream-chat-swift/runs/649869791?check_suite_focus=true

We're able to test iOS 11.4 and 12.4 (in parallel!), and we're using (we have to) backend integration test. The reported iOS12 crash was related to network request part, so I thought this makes sense to have!

Oh, also moved ClientLogger tests to Unit tests target since it was a unit test.

### CODE REVIEW

**Comment by buh** [comment]
Why did you select 11.4?
For example, our encoder/decoder has an issue with iOS under 11.2. `ISO8601DateFormatter` has `.withFractionalSeconds` only from 11.2+.

**Comment by b-onc** [comment]
@buh Latest version of each iOS SDK makes the most sense, we can't support all versions of iOS 11 and 12. Most users will be on the latest update of major version anyway. If they're not, the solution is simple -> Please update to latest iOS 11/12

File: fastlane/Fastfile [hunk 1]
```diff
@@ -135,6 +135,25 @@ lane :test_without_building do
   scan(project: "StreamChat.xcodeproj", scheme: "StreamChat", test_without_building: true, only_testing: ["StreamChatClientTests"])
 end
 
+desc "Runs integrations tests with backend. These tests make network connections so they're sometimes not reliable, hence we run them up to 3 times in case of failure"
+lane :test_backend_integration do |options|
+  @test_retries = 3
```
Comment (VojtaStavik) [inline fastlane/Fastfile:140]: 👍

**Comment by VojtaStavik** [comment]
@b-onc I realized we don't run unit tests, only integration tests. Is that on purpose?

**Comment by b-onc** [comment]
> @b-onc I realized we don't run unit tests, only integration tests. Is that on purpose?

That was on error, I've updated PR

**Pull request merged**